src/processor/operator/recursive_extend/recursive_join.cpp

```diff
@@ -189,6 +190,7 @@ void RecursiveJoin::computeBFS(ExecutionContext* context) {
         } else {
             // Otherwise move to the next frontier.
             bfsState->finalizeCurrentLevel();
+            scanFrontier->setNodePredicateExecFlag(bfsState->isLastLevel());
```

Comment:
This is not needed. Simply set Flag to False.

Reply:
Also remove `isLastLevel()`.